src/Umbraco.Core/Scoping/Scope.cs

```diff
@@ -348,6 +360,23 @@ public void Dispose()
 #endif
             }
 
+            // Decrement the lock counters on the parent if any.
+            if (ParentScope != null)
```

Comment:
I don't fully understand this check. If we are on the root Scope then nothing is decremented? Does that mean that if there are never child scopes, nothing is counted and we'll just keep making SQL lock calls for read/write locks requested only when there's an outer scope?